src/discussions/comments/data/slices.js

```diff
@@ -20,26 +20,31 @@ const commentsSlice = createSlice({
     // TODO: save in localstorage so user can continue editing?
     commentDraft: null,
     postStatus: RequestStatus.SUCCESSFUL,
-    pagination: {
-    },
+    pagination: {},
+    responsesPagination: {},
   },
   reducers: {
     fetchCommentsRequest: (state) => {
       state.status = RequestStatus.IN_PROGRESS;
     },
     fetchCommentsSuccess: (state, { payload }) => {
+      const { threadId } = payload;
+      // force endorsed to be null, true or false
+      const endorsed = payload.endorsed === undefined ? null : payload.endorsed;
```

Comment:
why are we assigning null to `endorsed`

Reply:
There are two types of posts, discussions and questions. 
Discussions are simple, and only track a single type of comment. 
Questions are more complex. They differentiate between endorsed and unendorsed comments. For questions, an endorsed comment is paginated separately, and displayed separately. 

So we need to track comments on the basis of endorsement: null if they are on a discussion, true/false if they on a question post and are endorsed/not-endorsed.